pkg/envoy/ads/stream.go

```diff
@@ -20,6 +22,42 @@ import (
 	"github.com/openservicemesh/osm/pkg/utils"
 )
 
+// This is the workqueue Job implementation to issue a Send Response to a proxy
+type sendJob struct {
+	// Send Response parametres
+	typeurls  mapset.Set
+	proxy     *envoy.Proxy
+	cfg       configurator.Configurator
+	adsStream *xds_discovery.AggregatedDiscoveryService_StreamAggregatedResourcesServer
+	request   *xds_discovery.DiscoveryRequest
+	xdsServer *Server
+
+	// Optional waiter
+	jobDoneChannel chan struct{}
+}
+
+// Run implementation for sendResponse job
+func (tj *sendJob) Run() {
+	err := (*tj.xdsServer).sendResponse(tj.typeurls, tj.proxy, tj.adsStream, tj.request, tj.cfg)
+	if err != nil {
+		log.Error().Err(err).Msgf("Failed to create and send %v update to Envoy with xDS Certificate SerialNumber=%s",
+			tj.typeurls, tj.proxy.GetCertificateCommonName().String())
+	}
+	tj.jobDoneChannel <- struct{}{}
+}
+
+// JobName implementation for this job, for logging purposes
+func (tj *sendJob) JobName() string {
+	return fmt.Sprintf("sendJob-%s", tj.proxy.GetCertificateSerialNumber())
```

Comment:
Do we intend to uniquely identify a job?
I'd add a timestamp and/or perhaps a `UID` field in the `sendJob` struct, which is created on init.

Reply:
I'd agree with all of it. Not sure if I'd push the changes in this patch as this is a change for the workerpool class, but I definitely agree. 
Potentially we could start talking transaction IDs, that by grepping on logs you see exactly what happened given a certain event.